app_remote/flaskr/templates/sidebar.html

```diff
@@ -1 +1,5 @@
-<body>Hi there, this is PII redaction app!</body>
\ No newline at end of file
+<body>
+    Hi there, this is PII redaction app!
+    <script src="https://static.zdassets.com/zendesk_app_framework_sdk/2.0/zaf_sdk.min.js"></script>
+    <script type="module" src="{{ url_for('static', filename='js/app.js') }}"></script>
```

Comment:
Ahhh! here is where it gets triggered. Ok in this case I would definitely load the function in the file and execute it in the script tag here. Easier testability. See earlier coment

Reply:
Exactly, that is something I was struggling when doing testing. Function was triggered in the file in which it got defined, this indeed caused issues and failed the test because one variable has got the chance to initialise. 

Instead, I've crated a separate JS file named `main.js` (inspired by this MDN example [here](https://github.com/mdn/js-examples/tree/master/modules/basic-modules)) and the function will be invoked there upon import of `main.js`. Tried to execute the function in the script tag also, but not working 😕. It looks like `<script type="module">` behaves very different from `<script>`.

```
<script type="module" src="forwardTicketId.js">
    console.log("This is the scope of forwardTicketId");  //this doesn't get print to console
    forwardTicketId(zafClient, appClient);  //this doesn't get triggered at all
   // no error was reported executing the above
</script>
```

Reply:
The type doesn't change from text/javascript for this one. There are only a few valid choices there https://www.geeksforgeeks.org/html-script-type-attribute/

Reply:
If we use `type=text/javascript ` to load `main.js`, it would cause
```
Uncaught SyntaxError: Cannot use import statement outside a module
```
According to [the html standard](https://html.spec.whatwg.org/multipage/scripting.html#attr-script-type), `module` is not an MIME type, so it is not appearing in the ''classic" types. A pic to illustrate the differences (copied from the standard) which I think it's quite intuitive to understand the behaviours

![image](https://user-images.githubusercontent.com/60019811/87369206-8554d000-c5b2-11ea-9a11-3655982a72ce.png)

Module is a relative new concept. Until recent years people start separating JS programs into modules and created many libraries and frameworks. What makes this one (the one used in this PR) special is this it's native.

Reply:
I learned something :) thanks @gabechu